sklearn/ensemble/_forest.py

```diff
@@ -507,53 +584,61 @@ def __init__(self,
             class_weight=class_weight,
             max_samples=max_samples)
 
-    def _set_oob_score(self, X, y):
-        """
-        Compute out-of-bag score."""
-        X = check_array(X, dtype=DTYPE, accept_sparse='csr')
-
-        n_classes_ = self.n_classes_
-        n_samples = y.shape[0]
-
-        oob_decision_function = []
-        oob_score = 0.0
-        predictions = [np.zeros((n_samples, n_classes_[k]))
-                       for k in range(self.n_outputs_)]
-
-        n_samples_bootstrap = _get_n_samples_bootstrap(
-            n_samples, self.max_samples
-        )
+    @staticmethod
+    def _get_oob_predictions(tree, X):
+        """Compute the OOB predictions for an individual tree.
 
-        for estimator in self.estimators_:
-            unsampled_indices = _generate_unsampled_indices(
-                estimator.random_state, n_samples, n_samples_bootstrap)
-            p_estimator = estimator.predict_proba(X[unsampled_indices, :],
-                                                  check_input=False)
+        Parameters
+        ----------
+        tree : DecisionTreeClassifier object
+            A single decision tree classifier.
+        X : ndarray of shape (n_samples, n_features)
+            The OOB samples.
 
-            if self.n_outputs_ == 1:
-                p_estimator = [p_estimator]
+        Returns
+        -------
+        y_pred : ndarray of shape (n_samples, n_classes, n_outputs)
+            The OOB associated predictions.
+        """
+        y_pred = tree.predict_proba(X, check_input=False)
+        y_pred = np.array(y_pred, copy=False)
+        if y_pred.ndim == 2:
+            # binary and multiclass
+            y_pred = y_pred[..., np.newaxis]
+        else:
+            # swap such that the `n_outputs` is the last axis
+            y_pred = np.rollaxis(y_pred, axis=0, start=3)
```

Comment:
:see_no_evil:

Reply:
Trust me, ["I'm an expert'](https://www.youtube.com/watch?v=BKorP55Aqvg) :)

Reply:
Maybe a more verbose comment would help the next person more then ;)

Reply:
Or alternatively using [numpy.transpose](https://numpy.org/doc/stable/reference/generated/numpy.transpose.html) would be more readable.